### ISSUE TITLE
[QUESTION] Get Request reference in ValidationPipe or provide some context for validation

### ISSUE THREAD

**Issue by mopcweb** [TypedProject/tsed#1103]
Hi!

The question is concerned with usage of groups from [validation groups](https://www.npmjs.com/package/class-validator#validation-groups) from `class-validator` lib.

I need to perform conditional IsOptional validation for create and update operations and unfortunately to make so without duplicating classes separately for each method it is impossible.

So example w/ 2 class, one of which is extra class (((
```ts
class UpdateEntity implements IEntity {
  @IsOptional()
  @IsNumber()
  public numberProperty?: number;

  @IsOptional()
  @IsString()
  public stringProperty?: string;
}

// In order not to create this extra class
class CreateEntity extends UpdateEntity implements IEntity {
  @IsNumber()
  public numberProperty: number;

  @IsString()
  public stringProperty: string;
}
```

And w/ groups
```ts
class UpdateEntity implements IEntity {
  @IsOptional({ groups: ['update'] })
  @IsNumber()
  public numberProperty?: number;

  @IsOptional({ groups: ['update'] })
  @IsString()
  public stringProperty?: string;
}
```

So basically question is for how to provide this context to ValidationPipe? 
One way is to use Request.method, i hope ...

Thanks

**Comment by Romakita** [comment]
Hello @mopcweb 
This feature doesn't exists but it's already identified  by this story #808 
This feature will be probably the next works for me :)

See you

**Comment by mopcweb** [comment]
@Romakita it would be super cool! Thanks )
